ssm/message_directory.py

```diff
@@ -0,0 +1,138 @@
+"""
+Copyright (C) 2018 STFC.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+   http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+
+@author: Greg Corbett
+"""
+
+import logging
+import os
+import uuid
+
+# logging configuration
+LOG = logging.getLogger(__name__)
+
+
+class MessageDirectory(object):
+    """A structure for holding Accounting messages in a directory."""
+
+    def __init__(self, path):
+        """Create a new directory structure for holding Accounting messages."""
+        self.directory_path = path
+
+    def add(self, data):
+        """Add the passed data to a new file and return it's name."""
+        # Use uuid4 to create a file name because uuid4 is
+        # sufficient to create an unique ID.
+        name = uuid.uuid4()
+
+        try:
+            # Open the file and write the provided data into the file.
+            with open("%s/%s" % (self.directory_path, name), 'w') as message:
+                message.write(data)
+        except (IOError, OSError) as error:
+            LOG.error("Could not create file %s/%s: %s",
+                      self.directory_path, name, error)
+
+        # Return the name of the created file as a string,
+        # to keep the dirq like interface.
+        return "%s" % name
+
+    def count(self):
+        """
+        Return the number of elements in the queue.
+
+        Regardless of their state.
+        """
+        return len(self._get_messages())
+
+    def get(self, name):
+        """Return the content of the named message."""
+        with open("%s/%s" % (self.directory_path, name)) as message:
+            content = message.read()
+        return content
+
+    def lock(self, _name):
+        """Return True to simulate a successful lock. Does nothing else."""
+        return True
+
+    def purge(self):
+        """
+        Do nothing, as there are no old/intermediate directories to purge.
+
+        Only included to preserve dirq interface.
+        """
+        LOG.debug("purge() called, but purge() does nothing.")
+
+    def remove(self, name):
+        """Remove the named message."""
+        try:
+            os.unlink("%s/%s" % (self.directory_path, name))
+        except (IOError, OSError) as error:
```

Comment:
I'm having second thoughts about this error handling as well, logging the error but carrying on regardless. 

[dirq seems like it would let the exception raise up](https://github.com/cern-mig/python-dirq/blob/75f63e193c209edd40a7a643386100d953bcbdab/dirq/QueueSimple.py#L311)